src/Core/Form/IdentifiableObject/CommandBuilder/Product/PricesCommandsBuilder.php

```diff
@@ -30,46 +30,55 @@
 
 use PrestaShop\PrestaShop\Core\Domain\Product\Command\UpdateProductPricesCommand;
 use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductShopConstraint;
+use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopId;
+use PrestaShop\PrestaShop\Core\Form\IdentifiableObject\CommandBuilder\Accessor\CommandAccessor;
+use PrestaShop\PrestaShop\Core\Form\IdentifiableObject\CommandBuilder\Accessor\CommandAccessorConfig;
+use PrestaShop\PrestaShop\Core\Form\IdentifiableObject\CommandBuilder\Accessor\CommandField;
 
 /**
  * Builder used to build UpdateProductPricesCommand
  */
-class PricesCommandsBuilder implements ProductCommandsBuilderInterface
+class PricesCommandsBuilder implements MultistoreProductCommandsBuilderInterface
 {
+    /**
+     * @var string
+     */
+    private $modifyAllNamePrefix;
+
+    /**
+     * @param string $modifyAllNamePrefix
+     */
+    public function __construct(string $modifyAllNamePrefix)
+    {
+        $this->modifyAllNamePrefix = $modifyAllNamePrefix;
+    }
+
     /**
      * {@inheritdoc}
      */
-    public function buildCommands(ProductId $productId, array $formData): array
+    public function buildCommands(ProductId $productId, array $formData, ShopId $shopId): array
     {
         if (!isset($formData['pricing'])) {
             return [];
         }
 
         $priceData = $formData['pricing'];
-        $command = new UpdateProductPricesCommand($productId->getValue());
+        $config = new CommandAccessorConfig($this->modifyAllNamePrefix);
+        $config
+            ->addMultiStoreField('[retail_price][price_tax_excluded]', 'setPrice', CommandField::TYPE_STRING)
+            ->addMultiStoreField('[retail_price][ecotax]', 'setEcotax', CommandField::TYPE_STRING)
+            ->addMultiStoreField('[tax_rules_group_id]', 'setTaxRulesGroupId', CommandField::TYPE_STRING)
+            ->addMultiStoreField('[on_sale]', 'setOnSale', CommandField::TYPE_STRING)
```

Comment:
`onSale` is a string, not a boolean?

Reply:
Yep, you're right! What is weird is that it was still working, which makes me think the `PropertyAccess` component probably handles casting internally